src/App.tsx

```diff
@@ -58,203 +57,99 @@ const PageNumber = styled.div`
-    }
-    case "page number changed": {
-      return {
-        ...state,
-        page: event.pagePayload.page,
-      };
-    }
-    case "select movie response set": {
-      return {
-        ...state,
-        selectedMovie: event.selMovieResPayload,
-      };
-    }
-  }
-};
-
 function App() {
-  const [state, update] = useReducer(reducer, initialState);
-  const [selectedID, updateSelectedID] = React.useState("");
-  const [popUpState, updatePopUpState] = React.useState(false);
-  const [mediaType, setMediaType] = React.useState("movie");
+  const [state, update] = useReducer(reducer, initialState());
 
   useEffect(() => {
-    async function callToAPIs() {
+    async function fetchMovies() {
       try {
-        const [movieResponse] = await Promise.all([
-          getMovies(state.search, state.page, mediaType),
-          getSelected(selectedID),
-        ]);
-        update({ type: "movie response set", movieResPayload: movieResponse });
+        // destructure and alias so Search can be referred to as movies
+        const { Search: movies } = await getMovies(
+          state.search,
+          state.page,
+          state.mediaType
+        );
+        const detailedMovies = await Promise.all(
+          movies.map((movie) => getDetailedMovie(movie.imdbID))
+        );
+        update({ type: "movies fetched", movies: detailedMovies });
       } catch (error) {
         console.log(error);
       }
     }
     if (state.search !== null) {
-      callToAPIs();
+      fetchMovies();
     }
-  }, [state.search, state.page, mediaType]);
-
-  useEffect(() => {
-    async function fetchData() {
-      const selMovieResponse = await getSelected(selectedID);
-
-      update({
-        type: "select movie response set",
-        selMovieResPayload: selMovieResponse,
-      });
-    }
-    fetchData();
-  }, [update, selectedID]);
+  }, [state.search, state.page, state.mediaType]);
 
   const sortedMovies = React.useMemo(() => {
-    return state.result
-      ? state.result.Search.slice(0).sort((a, b) => {
+    return state.movies
+      ? state.movies.slice(0).sort((a, b) => {
           return parseInt(a.imdbRating) - parseInt(b.imdbRating);
         })
       : [];
-  }, [state.result?.Search]);
+  }, [state.movies]);
 
-  return !state.result ? null : (
+  return !state.movies ? null : (
     <MainContainer>
       <Title>Movie Database</Title>
-      {popUpState ? (
+      {state.popupState && state.selectedMovie && state.movies.length ? (
         <PopUp
-          selected={state.selectedMovie}
-          backClick={(click) => {
-            updatePopUpState(click);
-            update({
-              type: "select movie response set",
-              selMovieResPayload: initialDetailedState,
-            });
-            updateSelectedID("");
-          }}
+          selected={state.movies.find(
+            (movie) => movie.imdbID === state.selectedMovie
+          )}
+          backClick={() =>
+            void update({
```

Comment:
Why do you put void ahead of all the updates?

Reply:
it just explicitly returns void from this callback as opposed to whatever update could return (it might not) but it's probably just my style and a bit overkill

Reply:
Ah okay, that makes sense